horreum-web/src/domain/actions/ActionComponentForm.tsx

```diff
@@ -223,6 +247,37 @@ export default function ActionComponentForm(props: ActionComponentFormProps) {
                     </FormGroup>
                 </>
             )}
+            {props.action.type === "slack-channel-message" && (
+                <>
+                    <SlackTokenInput
+                        helpIcon={<ExpressionHelp {...props} />}
+                        secrets={props.action.secrets}
+                        onChange={secrets => update({ secrets })}
+                    />
+                    <FormGroup label="Channel" labelIcon={<ExpressionHelp {...props} />} fieldId="channel">
+                        <TextInput
+                            id="channel"
+                            value={(props.action.config as SlackChannelMessage).channel}
+                            onChange={(_event, channel) => updateConfig({ channel })}
+                        />
+                    </FormGroup>
+                    <FormGroup label="Formatter" fieldId="formatter">
+                        <EnumSelect
+                            options={
+                                props.action.event === CHANGE_NEW
+                                    ? { changeToMarkdown: "Change to Markdown" }
+                                    : props.action.event === EXPERIMENT_RESULT_NEW
+                                      ? { experimentResultToMarkdown: "Experiment result to Markdown" }
+                                      : props.action.event === TEST_NEW
+                                        ? { testToSlack: "Test to Slack Markdown" }
+                                        : {}
```

Comment:
There's no formatter for `RUN_NEW`. Perhaps there should be. I decided that creating it was outside the scope of "implementing the slack action". (If affects GitHub issue comment, too, which allows selecting "run/new" while there's no applicable formatter, and the action can't actually be created.)

I find the implementation of the action form a bit awkward, and one of my complaints is this formatter popup when selecting the event really defines which formatter can apply anyway and it seems odd to require also selecting the formatter. I wrote up an issue about this, #1532

Reply:
I see two issues with the UI ... the formatters are one and another is the token that is sent as `******` to the server. since both are common to all the actions I say that we fix them in separate issues.

Reply:
if you can get all the formatters for slack to be present in this PR that would be appreciated